buildtest/buildsystem/base.py

```diff
@@ -276,6 +275,8 @@ def get_test_extension(self):
         shell = self.get_shell()
         if "python" in shell:
             return "py"
+        if "csh" in shell:
```

Comment:
I think the .sh extension is fine for csh as well, it's intended to be portable. See https://unix.stackexchange.com/questions/31760/file-extensions-for-unix-shell-scripts. If we do this, then we would need to honor all the other sub-flavors of shells, which doesn't add value. It's the interpreter at the top that matters.

Reply:
Probably makes sense to remove .csh feature if you agree. It adds extra work for buildtest to support multiple shells. I knew what we were getting into when adding support for csh

Reply:
I think it's okay to keep it :) I'm just saying for this function (that returns an extension) we can just return .sh.